neoq.go

```diff
@@ -255,6 +251,19 @@ func ConnectionString(connectionString string) ConfigOption {
 	}
 }
 
+// AddLogger configures neoq to use the spcified logger
+func AddLogger(logger Logger) ConfigOption {
```

Comment:
I really like the With function names. `WithLogger` in this case

Reply:
I've been avoiding it because the intended usage for these is through `WithConfig`, and when `With*` is used with another `With*`, it begins to stutter, e.g. 

`nq.WithConfig(WithLogger(logger))`

Which, now that I'm seeing in action, actually isn't too bad, but I think I'd actually prefer:

`nq.WithConfig(LoggerConfig(logger))` or something similar. 

Am I imagining the stutter and it's not a problem? Legitimate question. If I am, I think all the `ConfigOpt` functions should be standardized around as `WithX`.

Reply:
So, the WithOptions I feel would not be used quite often, since you can pass your collection of configurators to the  constructor. 

`neoq.New(..., WithLogger(someLogger), WithDeadline(tomorrow), WithConfurrency(oneWholeThing))`


That's all to say that if it stutters a bit when used like that it is probably okay.

Reply:
> So, the WithOptions I feel would not be used quite often, since you can pass your collection of configurators to the constructor.

What do you think of dropping `WithConfig` from the interface?

Reply:
I think that it's a good idea. Force users to set things up correctly in the beginning is never a bad thing.

Reply:
Cool, I'm hip to that :+1: